pkg/endpoint/log.go

```diff
@@ -41,6 +49,20 @@ func (e *Endpoint) getLogger() *logrus.Entry {
 	return (*logrus.Entry)(v)
 }
 
+// getPolicyLogger returns a logger to be used for policy update debugging, or nil,
+// if not configured.
+func (e *Endpoint) getPolicyLogger() *logrus.Entry {
+	v := atomic.LoadPointer(&e.policyLogger)
+	return (*logrus.Entry)(v)
+}
+
+// policyDebug logs the 'msg' with 'fields' if policy debug logging is enabled.
+func (e *Endpoint) policyDebug(fields logrus.Fields, msg string) {
+	if dbgLog := e.getPolicyLogger(); dbgLog != nil {
+		dbgLog.WithFields(fields).Debug(msg)
```

Comment:
Do we need to check if `fields` is not nil?

Reply:
Apparently not the test is passing. Iterating a nil map is fine.